hummingbot/market/kraken/kraken_api_order_book_data_source.py

```diff
@@ -0,0 +1,288 @@
+        """
+        async with aiohttp.ClientSession() as client:
+
+            trading_pairs_response = await client.get(ASSET_PAIRS_URL)
+            trading_pairs_response: aiohttp.ClientResponse = trading_pairs_response
+
+            if trading_pairs_response.status != 200:
+                raise IOError(f"Error fetching Kraken trading pairs. "
+                              f"HTTP status is {trading_pairs_response.status}.")
+
+            trading_pairs_data: Dict[str, Any] = await trading_pairs_response.json()
+
+            trading_pairs: Dict[str, Any] = {pair: {f"{k}Asset": trading_pairs_data["result"][pair][k] for k in ["base", "quote"]}
+                                            for pair in trading_pairs_data["result"]}
+            
+            trading_pairs_str: str = ','.join(trading_pairs.keys())
+
+            market_response = await client.get(f"{TICKER_URL}?pair={trading_pairs_str}")
+            market_response: aiohttp.ClientResponse = market_response
+
+            if market_response.status != 200:
+                raise IOError(f"Error fetching Kraken markets information. "
+                              f"HTTP status is {market_response.status}.")
+
+            market_data = await market_response.json()
+
+            market_data: List[Dict[str, Any]] = [{"pair": pair, **market_data["result"][pair], **trading_pairs[pair]}
+                                                for pair in market_data["result"]
+                                                if pair in trading_pairs]
+
+            # Build the data frame.
+            all_markets: pd.DataFrame = pd.DataFrame.from_records(data=market_data, index="pair")
+            all_markets["lastPrice"] = all_markets.c.map(lambda x: x[0])
+            quotes: Dict[str, List[str]] = {}
+            quotes["usdt"] = ["ETH", "XBT", "DAI", "USDC"]
+            quotes["usdt_r"] = ["ZUSD", "EUR", "CAD", "GBP"]
+            quotes["xeth_r"] = ["ZJPY"]
+            quotes["eth_r"] = ["CHF"]
+            quotes_all: List[str] = chain([quotes[currency] for currency in quotes])
+            price: Dict[str, float] = {}
+            for quote in quotes["usdt"]:
+                price[quote] = float(all_markets.loc[f"{quote}USDT"].lastPrice)
+            for quote in quotes["usdt_r"]:
+                price[quote] = 1./float(all_markets.loc[f"USDT{quote}"].lastPrice)
+            for quote in quotes["eth_r"]:
+                price[quote] = price["ETH"]/float(all_markets.loc[f"ETH{quote}"].lastPrice)
+            for quote in quotes["xeth_r"]:
+                price[quote] = price["ETH"]/float(all_markets.loc[f"XETH{quote}"].lastPrice)
+            usd_volume: float = [
+                (
+                    quoteVolume * price[quote] if trading_pair[-3:] in quotes_all else
+                    quoteVolume * price[quote] if trading_pair[-4:] in quotes_all else
+                    quoteVolume
+                )
```

Comment:
It seems like for some trading pairs, the USD volume is not correctly calculated. In the below example,
The `USDVolume` for `SCEUR` should be ~USD100,000
![image](https://user-images.githubusercontent.com/6755922/75293179-89867980-57da-11ea-9147-0e8f150e9f69.png)

Reply:
ah, looks like not everything is as it appears on the trading webapp – will fix

Reply:
Looking at the API documentation and the trading webapp, it seems like the `volume` that is provided here is the base volume.